connectionmanager/exponentialbackoff.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 Northern.tech AS
+//
+//	Licensed under the Apache License, Version 2.0 (the "License");
+//	you may not use this file except in compliance with the License.
+//	You may obtain a copy of the License at
+//
+//	    http://www.apache.org/licenses/LICENSE-2.0
+//
+//	Unless required by applicable law or agreed to in writing, software
+//	distributed under the License is distributed on an "AS IS" BASIS,
+//	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//	See the License for the specific language governing permissions and
+//	limitations under the License.
+
+package connectionmanager
+
+import (
+	"math/rand"
+	"time"
+)
+
+type expBackoff struct {
+	attempts    int
+	exceededMax bool
+	// The maximum interval before it starts increasing linearly
+	maxInterval time.Duration
+	// Retry time does not increaseb beyond maxBackoff
+	maxBackoff time.Duration
+	// Smallest backoff sleep time
+	smallestUnit time.Duration
+}
+
+func (a *expBackoff) GetExponentialBackoffTime() time.Duration {
+	var nextInterval time.Duration
+	const perIntervalAttempts = 3
+
+	interval := 1 * a.smallestUnit
+	nextInterval = interval << (a.attempts / perIntervalAttempts)
```

Comment:
The `perIntervalAttempts` seems a bit arbitrary. I know this comes from the mender client, but I do not understand the purpose of such a mechanism.